packages/core/src/tracing/trace.ts

```diff
@@ -0,0 +1,65 @@
+import type { TransactionContext } from '@sentry/types';
+import { isThenable } from '@sentry/utils';
+
+import { getCurrentHub } from '../hub';
+import type { Span } from './span';
+
+/**
+ * Wraps a function with a transaction/span and finishes the span after the function is done.
+ *
+ * This function is meant to be used internally and may break at any time. Use at your own risk.
+ *
+ * @internal
+ * @private
+ */
+export function trace<T>(
+  context: TransactionContext,
+  callback: (span: Span) => T,
```

Comment:
super-l: WDYT about renaming `callback` to something like `traceTarget` / `traceSubject` or something along these lines? The way I understand this function, we want to start a txn or a span for this callback so I'd say we should give it a good name. `callback` sounds a little generic to me.  
But ofc we can (and probably will) iterate on this function in the future, so really, feel free to disregard this comment.

Reply:
I'm going to use callback for now since this is the language we are using for the cross SDK discussions about this, but you do make a good point. I'll come back and re-visit this after we chat about it as a group.